Add topology spread constraints to control how Pods are spread

dschunack opened the pull request:
#### Which component this PR applies to?

This PR applies to helm chart. It adds the possibility to use topologySpreadConstraints in the helm chart.

Tested one two EKS with k8s 1.21. 

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

topologySpreadConstraints is not supported yet by the helm chart, but we need it to spread pods over Zones.

#### Which issue(s) this PR fixes:

Fixes #4577

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

Added a line in README.MD and also an example in the values.yaml.

dschunack commented:
Hi,

maybe you can review this small PR. It's only a small modification of the helm cahrt, but I think it helps a lot if you have a multi AZ environment.

@gjtempleton @bskiba

steveizzle commented:
i would also appreciate this feature

gjtempleton commented on charts/cluster-autoscaler/Chart.yaml:
```diff
@@ -17,4 +17,4 @@ name: cluster-autoscaler
   - https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
 type: application
-version: 9.12.0
+version: 9.12.1
```
As we're adding new functionality rather than patching a bug, I'd prefer to make this a minor version bump.
```suggestion
version: 9.13.0
```

dschunack replied:
sure, I will do this later today

dschunack replied:
It's now Chart Version 9.13.0 and squashed

gjtempleton reviewed (changes requested):
Thanks for the PR.

One change and this should be good to go, however, could you please also squash down the commits to a single commit?

For the future, if you're looking to get PRs reviewed faster in the future, you're best to use the `/assign` command mentioned by the k8s-ci-robot bot's comment.

gjtempleton commented:
/assign @gjtempleton

gjtempleton commented:
Thanks @dschunack, can you please squash this down to a single commit and we can get it merged.

dschunack commented:
/retest

dschunack commented:
Hi @gjtempleton,

So, it's now squashed and rebased to a single commit. Sorry, for all the commits yesterday, Forgot to rebase it with the upstream origin master.

gjtempleton commented:
No problem, thanks for persisting!
/lgtm
/approve
